supplementary/alica_ros1/alica_ros_proxy/src/communication/AlicaRosCommunication.cpp

```diff
@@ -80,8 +80,7 @@ AlicaRosCommunication::~AlicaRosCommunication()
     _planTreeInfoSubscriber.shutdown();
     _syncReadySubscriber.shutdown();
     _syncTalkSubscriber.shutdown();
-    _rosNode->shutdown();
-    delete _rosNode;
+    _nh.shutdown();
```

Comment:
since this is not a pointer anymore, is this necessary?

Reply:
`delete` also calls `destructor`, in other words it's not related to this question
but calling `shutdown` prevents callback invocation in destroyed object

Reply:
I guess we need either this, the subscriber shutdowns, or to be careful with declaration order (and I'd rather not rely on the last) but we might not need both this and the subscriber explicit shutdowns